app/widgets/Roster/Roster.php

```diff
@@ -64,6 +69,34 @@ function onRoster()
         $this->onUpdate();
     }
 
+    function onIqGatewayGet($packet)
+    {
+        $this->rpc(
+            'Roster.addGatewayPrompt',
+            $packet->from,
+            ''.$packet->content->prompt,
+            ''.$packet->content->desc
+        );
+    }
+
+    function onIqGatewaySet($packet)
+    {
+        $form = json_decode($packet->content->extra);
```

Comment:
after other changes, this no longer happens, it seems.  So I've removed the json encoding again